src/main/java/ru/practicum/shareit/item/model/Item.java

```diff
@@ -1,16 +1,46 @@
 package ru.practicum.shareit.item.model;
 
-import lombok.Data;
+import jakarta.persistence.CollectionTable;
+import jakarta.persistence.Column;
+import jakarta.persistence.ElementCollection;
+import jakarta.persistence.Entity;
+import jakarta.persistence.FetchType;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.JoinColumn;
+import jakarta.persistence.ManyToOne;
+import jakarta.persistence.Table;
 import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.ToString;
 import ru.practicum.shareit.request.model.ItemRequest;
 
-@Data
-@EqualsAndHashCode(exclude = "available")
+import java.util.ArrayList;
+import java.util.List;
+
+@Entity
+@Table(name = "items")
+@Getter
+@Setter
+@ToString
+@EqualsAndHashCode(exclude = {"available", "request"})
```

Comment:
Сейчас в `equals()` попадают все поля, кроме `available` и `request`, включая `name` и `description`. Это может привести к неожиданному поведению при использовании коллекций (Set, Map), особенно если поля меняются.
В JPA-сущностях хорошей практикой является определение `equals()` и `hashCode()` только по `id`, желательно с `@EqualsAndHashCode.Include` и `onlyExplicitlyIncluded = true`
```suggestion
// остальные поля
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
public class Item {

    @EqualsAndHashCode.Include
    @Id
    private Long id;

    // остальные поля
}
```

Reply:
сделано